### DISCUSSION TITLE
Minimum feasible processing time

### DISCUSSION THREAD

**Discussion by Shellfishgene** [tphakala/birdnet-go#22] [General]
Hi!

So I had an extra "Orange Pi Zero" lying around, and thought maybe it's just powerful enough to run birdnet-go. Took me quite long to get tf-lite to compile (it's an ARM 7 32 bit CPU), but now it works. However the processing time for 3 sec segments is about 2300-2500 ms. Is anything below 3000 ms still ok? It seems to react quite slow, I wonder if it's getting behind on analyzing?

Cheers

**Comment by tphakala** [comment]
Hi and thanks for chiming in!

In current code there is a sliding window buffer which processes audio data in 1.5 second chunks to increase detection success rate, because of this processing times should be less than 1500ms to keep up. 

This can be changed for lower spec systems:

internal/myaudio/buffers.go line 19, change "var overlapSize int = 144000" to "var overlapSize int = 0" to disable sliding window.

With this change your hardware should keep up with realtime processing.

**Comment by Shellfishgene** [comment]
Great, that seems to work, thanks! The code looks very clean and well commented.
Do you think it would be possible to make birdnet-go into an Android app? I always thought all those old smartphones everyone has would make a good platform, they have a good microphone already and with mobile data you could put them somewhere without wifi.

**Reply by tphakala** [reply]
No I have no plans to do a mobile app, but I do have plans to implement some options which allows BirdNET-Go to run as battery powered remote station.

**Reply by unosonic** [reply]
hm ;) what prevents you (and me) to just connect a powerbank and modem to the birdnet-go-pi and throw it somewhere in the bush?
In fact that's what I've done with birdNet-Pi4 quite often. ok, have to say that the PI4 opens an OpenVPN connection to my server, for cmd & ctrl. So, curious, what sort of field option would that be?
